analyzers/src/SonarAnalyzer.Common/Rules/Utilities/TokenTypeAnalyzerBase.cs

```diff
@@ -80,130 +106,88 @@ protected abstract class TokenClassifierBase
-                this.token = token;
                 this.semanticModel = semanticModel;
                 this.skipIdentifiers = skipIdentifiers;
             }
 
-            public IEnumerable<TokenTypeInfo.Types.TokenInfo> Spans
-            {
-                get
-                {
-                    spans.Clear();
-                    ClassifyToken();
-
-                    foreach (var trivia in token.LeadingTrivia)
-                    {
-                        ClassifyTrivia(trivia);
-                    }
-
-                    foreach (var trivia in token.TrailingTrivia)
-                    {
-                        ClassifyTrivia(trivia);
-                    }
-
-                    return spans;
-                }
-            }
-
-            private void CollectClassified(TokenType tokenType, TextSpan span)
-            {
-                if (string.IsNullOrWhiteSpace(token.SyntaxTree.GetText().GetSubText(span).ToString()))
-                {
-                    return;
-                }
-
-                spans.Add(new TokenTypeInfo.Types.TokenInfo
+            private TokenTypeInfo.Types.TokenInfo TokenInfo(SyntaxToken token, TokenType tokenType) =>
+                string.IsNullOrWhiteSpace(token.ValueText)
+                ? null
+                : new()
                 {
                     TokenType = tokenType,
-                    TextRange = GetTextRange(Location.Create(token.SyntaxTree, span).GetLineSpan())
-                });
-            }
+                    TextRange = GetTextRange(token.GetLocation().GetLineSpan()),
+                };
 
-            private void ClassifyToken()
-            {
-                if (IsKeyword(token))
+            public TokenTypeInfo.Types.TokenInfo ClassifyToken(SyntaxToken token) =>
+                token switch
                 {
-                    CollectClassified(TokenType.Keyword, token.Span);
-                }
-                else if (IsStringLiteral(token))
-                {
-                    CollectClassified(TokenType.StringLiteral, token.Span);
-                }
-                else if (IsNumericLiteral(token))
-                {
-                    CollectClassified(TokenType.NumericLiteral, token.Span);
-                }
-                else if (IsIdentifier(token) && !skipIdentifiers)
-                {
-                    ClassifyIdentifier();
-                }
-            }
+                    _ when IsKeyword(token) => TokenInfo(token, TokenType.Keyword),
+                    _ when IsStringLiteral(token) => TokenInfo(token, TokenType.StringLiteral),
+                    _ when IsNumericLiteral(token) => TokenInfo(token, TokenType.NumericLiteral),
+                    _ when IsIdentifier(token) && !skipIdentifiers => ClassifyIdentifier(token),
+                    _ => null,
+                };
 
-            private void ClassifyIdentifier()
+            private TokenTypeInfo.Types.TokenInfo ClassifyIdentifier(SyntaxToken token)
```

Comment:
Can be turned into a switch expression (similar to `ClassifyToken`).

Reply:
That method is causing a lot of performance trouble and will be reworked in the next iteration. Let's keep it for the moment.